templates/xcassets/swift3.stencil

```diff
@@ -3,23 +3,33 @@
 
 {% if catalogs %}
 {% set enumName %}{{param.enumName|default:"Asset"}}{% endset %}
+{% set arResourceGroupType %}{{param.arResourceGroupTypeName|default:"ARResourceGroupAsset"}}{% endset %}
 {% set colorType %}{{param.colorTypeName|default:"ColorAsset"}}{% endset %}
 {% set dataType %}{{param.dataTypeName|default:"DataAsset"}}{% endset %}
 {% set imageType %}{{param.imageTypeName|default:"ImageAsset"}}{% endset %}
-{% set colorAlias %}{{param.colorAliasName|default:"AssetColorTypeAlias"}}{% endset %}
-{% set imageAlias %}{{param.imageAliasName|default:"AssetImageTypeAlias"}}{% endset %}
 {% set forceNamespaces %}{{param.forceProvidesNamespaces|default:"false"}}{% endset %}
 {% set accessModifier %}{% if param.publicAccess %}public{% else %}internal{% endif %}{% endset %}
 #if os(macOS)
-  import AppKit.NSImage
-  {{accessModifier}} typealias {{colorAlias}} = NSColor
-  {{accessModifier}} typealias {{imageAlias}} = NSImage
-#elseif os(iOS) || os(tvOS) || os(watchOS)
-  import UIKit.UIImage
-  {{accessModifier}} typealias {{colorAlias}} = UIColor
-  {{accessModifier}} typealias {{imageAlias}} = UIImage
+  import AppKit
+#elseif os(iOS)
+{% if resourceCount.arresourcegroup > 0 %}
+  import ARKit
+{% endif %}
+  import UIKit
+#elseif os(tvOS) || os(watchOS)
+  import UIKit
 #endif
 
+// Deprecated typealiases
+{% if resourceCount.color > 0 %}
+@available(*, deprecated, message: "This typealias will be removed in SwiftGen 7.0")
```

Comment:
Shouldn't we use `@available(*, renamed, …)` instead of `deprecated`? This makes Xcode provide a nice fix-it

Reply:
Uhm, it becomes something like `ColorAsset.Color`. Not sure if we want to have people actually use those type(alias)es? They're an internal artefact of having cross-platform code.

But yeah, I can switch to `renamed`.

Reply:
Well people weren't supposed to directly use that root typealias in the first place even before that namespacing so it always was an internal artefact of cross platform anyway. It should be pretty rare to find users using those typealiases and who would require them to rename. But for those very rare who do…